kivy/graphics/vertex_instructions_line.pxi

```diff
@@ -314,8 +318,8 @@ cdef class Line(VertexInstruction):
             indices_count += 12
             vertices_count += 4
         elif cap == LINE_CAP_ROUND:
-            indices_count += (self._cap_precision * 3) * 2
-            vertices_count += (self._cap_precision) * 2
+            indices_count += 2 * (self._cap_precision + 1) * 3 * 2
+            vertices_count += 2 * (self._cap_precision + 1) * 2
```

Comment:
@tshirtman This is *not* the right approach. If you look in the docs, you'll see what [`cap_precision`](https://kivy.org/docs/api-kivy.graphics.html#kivy.graphics.Line.cap_precision) is documented as and this line basically invalidates everything in there, no? The same would apply for the indicies as it's bound to the vertices.

Basically what I'm trying to say:

![image](https://user-images.githubusercontent.com/8825439/43490586-98205b28-9521-11e8-9e16-7ccf6d913ee3.png)

Let's say the "cap" is a semicircle and let's take the `cap_precision == 4` for simplicity. The red lines in the previous implementation would represent this part:

> Number of iteration for drawing the "round" cap [...]

Therefore the total count of points/vertices should be `self._cap_precision - 1` for the semicircle i.e. the line 599 or one of these: `2 * cap_precision` or `2 * cap_precision - 1` depending on whether the center point should be duplicated on both ends or if it takes the point from the already existing line segment. I believe _**`4`**_` * (cap_precision + 1)` is bad approach. That would also make the indices `vertices_count * 3`.

The PR author pretty much stated he draws the full circles, but it seems strangely wasteful and instead I believe the wrong thing (perhaps except +/- 1 in the numbers, but if it worked before, don't fix it) is actually the starting angle ([picture](https://math.stackexchange.com/a/1201367/320970)). 

```python
>>> from math import atan2, degrees
>>> degrees(atan2(0, 1))
0.0
>>> degrees(atan2(1, 0))
90.0
>>> degrees(atan2(0.5, 0.5))
45.0
>>> degrees(atan2(0.3, 0.7))
23.19859051364819
```

I haven't really poked in the code, I only read it, however when I draw the line from the example in the issue up ↔ down, the start angle is either `0`(most likely?) or `90`, when I draw left → right the line is correct and when I draw right → left, the caps are hidden in the line itself (good angle, bad direction because we default to counterclockwise drawing for that).

Basically the PR "fixes" only the "looks" not the "brains" and works only because you can't see the wrong direction if you overdraw the only part that actually shows that the direction is wrong (the diameter segment).

Reply:
Yes, i agree the fix is crude, and thanks for the writeup, i'm looking again at it, and hope to fix it proper.